monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/commons/TableRowCreator.js

```diff
@@ -64,7 +90,178 @@ function ProxyRowCreator(props) {
             case 'wsdlUrl':
-                return <TableCell>{data.nodes.map(node=><WsdlCell data={node}/>)}</TableCell>
+                return <TableCell>{data.nodes.map(node=><StringCell data={node.wsdlUrl}/>)}</TableCell>
+            default:
+                <TableCell>Table data not available</TableCell>
+        }})}
+    </TableRow>
+}
+
+function EndpointRowCreator(props) {
+    const { headers, data } = props;
+    return <TableRow>
+        {headers.map(header => {switch(header.id) {
+            case 'name':
+                return <TableCell>{data.name}</TableCell>
+            case 'nodes':
+                return <TableCell><table>{data.nodes.map(node=><NodesCell data={node} proxyName={data['service']}/>)}</table></TableCell>
+            case 'type':
+                return <TableCell><table>{data.nodes.map(node=><StringCell data={node.type} proxyName={data['service']}/>)}</table></TableCell>
+            case 'state':
+                return <TableCell>{data.nodes.map(node=><SwitchStatusCell data={node.state}/>)}</TableCell>
+            default:
+                <TableCell>Table data not available</TableCell>
+        }})}
+    </TableRow>
+}
+
+function InboundEndpointRowCreator(props) {
+    const { headers, data } = props;
+    return <TableRow>
+        {headers.map(header => {switch(header.id) {
+            case 'name':
+                return <TableCell>{data.name}</TableCell>
+            case 'nodes':
+                return <TableCell><table>{data.nodes.map(node=><NodesCell data={node} proxyName={data['service']}/>)}</table></TableCell>
+            case 'protocol':
+                return <TableCell><table>{data.nodes.map(node=><StringCell data={node.protocol} proxyName={data['service']}/>)}</table></TableCell>
+            case 'port':
+                return <TableCell>{data.nodes.map(node=><StringCell data={node.port}/>)}</TableCell>
+            default:
+                <TableCell>Table data not available</TableCell>
+        }})}
+    </TableRow>
+}
+
+function MessageStoreRowCreator(props) {
+    const { headers, data } = props;
+    return <TableRow>
+        {headers.map(header => {switch(header.id) {
+            case 'name':
+                return <TableCell>{data.name}</TableCell>
+            case 'nodes':
+                return <TableCell><table>{data.nodes.map(node=><NodesCell data={node} proxyName={data['service']}/>)}</table></TableCell>
+            case 'type':
+                return <TableCell><table>{data.nodes.map(node=><StringCell data={node.type} proxyName={data['service']}/>)}</table></TableCell>
+            case 'message_count':
+                return <TableCell>{data.nodes.map(node=><StringCell data={node.message_count}/>)}</TableCell>
+            default:
+                <TableCell>Table data not available</TableCell>
+        }})}
+    </TableRow>
+}
+function APIRowCreator(props) {
```

Comment:
Can't we generalize these TableRow component generation functions ? Like, 
```function tableRowGenerator(props) ;```
In the props it would get the headers. Then it would have this switch case with all the cases.
Then in other functions that are calling it can simply filter the list of applying headers and then call that function to generate the TableRow component.